src/content/docs/en/recipes/build-forms-api.mdx

```diff
@@ -133,187 +165,119 @@ This recipe shows you how to send form data to an API endpoint and handle that d
 
     </UIFrameworkTabs>
 
-  2. Create the `POST` API endpoint that will receive the form data. Use `request.formData()` to process it. Be sure to validate the form values before you use them. 
+
+3. Now, create a function that accepts a submit event, then pass it as a `submit` handler to your form. In the function, call `preventDefault` on the event to override the browser's default submission process. 
   
-        This example sends a JSON object with a message back to the client.
-
-      ```ts title="src/pages/api/feedback.ts" "request.formData()" "post"
-      import type { APIRoute } from "astro";
-
-      export const POST: APIRoute = async ({ request }) => {
-        const data = await request.formData();
-        const name = data.get("name");
-        const email = data.get("email");
-        const message = data.get("message");
-        // Validate the data - you'll probably want to do more than this
-        if (!name || !email || !message) {
-          return new Response(
-            JSON.stringify({
-              message: "Missing required fields",
-            }),
-            { status: 400 }
+    Then, create a `FormData` object and send it in a `POST` request to your endpoint using `fetch`. 
+
+    <UIFrameworkTabs>
+      <Fragment slot="preact">
+        ```tsx title="src/components/FeedbackForm.tsx" "/api/feedback" add={1, 4-17, 34} add="onSubmit={submit}"
```

Comment:
```suggestion
        ```tsx title="src/components/FeedbackForm.tsx" "/api/feedback" add={1, 4-17, 34} add="onSubmit={submit}" "formData" "e.preventDefault();"
```

Reply:
Shall I do this for all the frameworks?

Reply:
If you think it looks good, then yeah!